packages/salesforcedx-utils-vscode/src/telemetry/telemetry.ts

```diff
@@ -52,25 +57,38 @@ export class TelemetryBuilder {
   }
 }
 
+// export only for unit test
+export class TelemetryServiceProvider {
+  public static instances = new Map<string, TelemetryService>(); // public only for unit test
+  public static getInstance(extensionName?: string): TelemetryService {
+    // default if not present
+    const name = extensionName || SFDX_CORE_EXTENSION_NAME;
+    if (!TelemetryServiceProvider.instances.has(name)) {
+      TelemetryServiceProvider.instances.set(name, new TelemetryService());
+    }
+    return TelemetryServiceProvider.instances.get(name)!;
```

Comment:
```suggestion
    return TelemetryServiceProvider.instances.get(name);
```

if an instance with the name is not found, it is created and put into the map, and then retrieved and returned.  There should not be any time when this method needs to return undefined, which is what I think the "!" at the end allows.  Can the "!" be removed here?

Reply:
thanks for pointing this out. There is strict null checking for typescript in the project so it has to be added in order to bypass the rule.